### PR TITLE
Redis

### PR DESCRIPTION
**Pull request by leighdice**
#9

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -8,11 +9,21 @@ class VenueApp < Sinatra::Base
...
   helpers Errors, RequestTimer
   File.open('venue_app.pid', 'w') {|f| f.write Process.pid }
+  set :show_exceptions, false
 
   configure :development do
     enable :logging, :dump_errors, :run, :sessions
     Mongoid.load!(File.join(File.dirname(__FILE__), 'config/mongoid.yml'), :development)
-    Mongoid.raise_not_found_error = false    
+    Mongoid.raise_not_found_error = false
+    $redis = Redis.new(:host => "127.0.0.1", :port => 6379)
```
Comment (SimonRichardson) [inline app.rb]: Env variables need to be next on the list?

File: app.rb [hunk 2]
```diff
@@ -50,27 +66,38 @@ class VenueApp < Sinatra::Base
   # get venue by id
   get '/venues/:id' do
     t = request_timer_start
-    venue = Venue.find(params[:id])
 
-    return error_not_found(params[:id]) if venue.nil?
+    # Attempt to fetch from redis
+    # If key does not exist, fetch from db and create new key
+    if $redis.exists(params[:id])
+      venue = JSON.parse($redis.get(params[:id]))
```
Comment (SimonRichardson) [inline app.rb]: Would be great if we could just rendering it to the records without the need to parse it then unparse it again.

File: app.rb (hunk 1 above)
Comment (leighdice) [inline app.rb]: yeah, ill can use env variables in `configure :production`

or do something like

```
Redis.new(:host => ENV['REDIS_HOST']  || 127.0.0.1, :port => ENV['REDIS_PORT'] || 6379)
```

File: app.rb (hunk 2 above)
Comment (leighdice) [inline app.rb]: aight ill see what i can do

**Pull request merged**